// react-glitch-effect/core/GlitchSquiggly/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _GlitchBase = _interopRequireDefault(require("../__core__/components/GlitchBase"));

var _SvgFilters = _interopRequireDefault(require("./components/SvgFilters"));

require("./style/index.css");

var GLITCH_ID = 'glitch-squiggly';

var GlitchSquiggly = function GlitchSquiggly(_ref) {
  var _ref$baseFrequency = _ref.baseFrequency,
      baseFrequency = _ref$baseFrequency === void 0 ? 0.02 : _ref$baseFrequency,
      children = _ref.children,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? 340 : _ref$duration,
      _ref$scaleNoise = _ref.scaleNoise,
      scaleNoise = _ref$scaleNoise === void 0 ? 5 : _ref$scaleNoise,
      rest = (0, _objectWithoutProperties2.default)(_ref, ["baseFrequency", "children", "duration", "scaleNoise"]);
  return /*#__PURE__*/_react.default.createElement(_GlitchBase.default, (0, _extends2.default)({
    duration: duration,
    glitchId: GLITCH_ID
  }, rest), /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, children), /*#__PURE__*/_react.default.createElement(_SvgFilters.default, {
    baseFrequency: baseFrequency,
    scaleNoise: scaleNoise
  }));
};

var _default = /*#__PURE__*/(0, _react.memo)(GlitchSquiggly);

exports.default = _default;
GlitchSquiggly.propTypes = {
  baseFrequency: _propTypes.default.number,
  children: _propTypes.default.node.isRequired,
  disabled: _propTypes.default.bool,
  duration: _propTypes.default.number,
  iterationCount: _propTypes.default.string,
  onHover: _propTypes.default.bool,
  onMouseEnter: _propTypes.default.func,
  onMouseLeave: _propTypes.default.func,
  scaleNoise: _propTypes.default.number
};
